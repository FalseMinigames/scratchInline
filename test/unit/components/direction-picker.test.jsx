import React from 'react';
import {componentWithIntl} from '../../helpers/intl-helpers';
import RotationStyles from '../../../src/components/direction-picker/direction-picker';
import DirectionPicker from '../../../src/components/direction-picker/direction-picker';
import PropTypes from 'prop-types';
import {intlShape} from 'react-intl';

describe('DirectionPicker', () => {

    const getComponent = function (props = {}) {
        return <DirectionPicker {...props} />;
    };

    test('matches snapshot', () => {
        const props = {
            // direction: PropTypes.number,
            disabled: false,
            // intl: intlShape,
            // labelAbove: PropTypes.bool,
            onChangeDirection: jest.fn(),
            onClickAllAround: jest.fn(),
            onClickDontRotate: jest.fn(),
            onClickLeftRight: jest.fn(),
            onClosePopover: jest.fn(),
            onOpenPopover: jest.fn(),
            popoverOpen: false
            // rotationStyle: PropTypes.string
        };


        const component = componentWithIntl(getComponent(props));
        const picker = component.root.findByType(DirectionPicker);
        console.log(picker);
        expect(component.toJSON()).toMatchSnapshot();
    });
});
