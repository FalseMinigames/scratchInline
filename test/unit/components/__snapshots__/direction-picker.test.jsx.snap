// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DirectionPicker matches snapshot 1`] = `
<label
  className={undefined}
>
  <span
    className={undefined}
  >
    <span>
      Direction
    </span>
  </span>
  <input
    className="undefined"
    disabled={false}
    label={
      <FormattedMessage
        defaultMessage="Direction"
        id="gui.SpriteInfo.direction"
        values={Object {}}
      />
    }
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyPress={[Function]}
    onSubmit={[Function]}
    tabIndex="0"
    type="text"
    value={undefined}
  />
</label>
`;
